ccm_web/server/src/init.ts

```diff
@@ -0,0 +1,22 @@
+import { NestFactory } from '@nestjs/core'
+
+import { AppModule } from './app.module'
+
+import baseLogger from './logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+/*
+Running this before clustering ensures that ltijs database tables have been set up initially
+and that the configuration is valid.
+*/
+async function bootstrap (): Promise<void> {
+  const app = await NestFactory.createApplicationContext(AppModule, {
+    logger: ['error', 'warn']
+  })
+  await app.close()
+}
+
+bootstrap()
+  .then(() => logger.info('ltijs has been initialized and configuration is valid.'))
+  .catch((e) => logger.error('An error occurred while initializing: ', e))
```

Comment:
I keep going back and forth about whether this makes sense. Maybe there's an alternative I'm not thinking of. I'm trying to avoid the case of either 1) `ltijs` migrations haven't been set up before clustering, and the setup process errors out in the threads that lag behind the first and 2) the configuration is invalid, and all the threads emit repetitive errors. Maybe neither of those scenarios is that big of a deal (the first only happens with an empty database, the second when the configuration is wrong, and it should error out, just not for each thread ideally). Still it seems somewhat logical to try to start one application successfully before starting several. I didn't see any `pm2` options like this, though you could probably do it with a few `pm2` commands (e.g. `start`, `scale`).

Even assuming this is a good idea, I wasn't exactly sure how to implement it.  I chose `createApplicationContext` -- see NestJS [standalone applications](https://docs.nestjs.com/standalone-applications) -- because it allowed for a more bare-bones start and the network listeners didn't seem necessary for these two checks. However, maybe it's better to use all the same middleware configurations that are in the `main.ts` `bootstrap` function. Alternatively, I could be even more minimal and selective, and call the `validateConfig` function directly from `config.ts` and only `createApplicationContext` with `LTIModule` (using a similar approach, though I'd need to play around with how it could be configured separately, but pretty sure it's possible).

Sorry, maybe this is confusing or too much detail, but I thought I'd lay it all out in case someone was interested and had an opinion. What I currently have "works", but may not be optimal for one or more reasons (listed above or not).

Reply:
@ssciolla Thanks for additional details. I really don't think I have a opinion. Lets keep this way and take it as we go. Thanks for thinking through on this.